cips/cip-26.md

```diff
@@ -0,0 +1,64 @@
+| cip | 26 |
+| - | - |
+| title | Reduce block time to six seconds |
+| description | Proposal to reduce block time on celestia-app to 6 seconds, from 12. |
+| author | Josh Stein ([@jcstein](https://github.com/jcstein)), Rootul Patel ([@rootulp](https://github.com/rootulp))| Sanaz Taheri ([@staheri14](https://github.com/staheri14) )
+| discussions-to | <https://forum.celestia.org/t/cip-decrease-block-time-to-6-seconds/1836> |
+| status | Draft |
+| type | Standards Track |
+| category | Core |
+| created | 2024-10-09 |
+
+## Abstract
+
+This CIP proposes to reduce the block time on celestia-app to 6 seconds, from 12. This will double the throughput and reduce the time it takes for transactions to be finalized in half. This CIP also proposes to increase the `ttl-num-blocks` parameter in the mempool configuration from 5 to 12 to maintain consistency with the new block time.
+
+## Motivation
+
+The motivation for this CIP stems from a discussion in Core Devs Call 17, where it was proposed to reduce the block time to 6 seconds from 12 seconds.
+
+## Specification
+
+1. The block time in celestia-app SHOULD be reduced from 12 seconds to 6 seconds. Concretely, this implies decreasing `TimeoutCommit` to 4.2 seconds and `TimeoutPropose` to 3.5 seconds.
+    1. The `TimeoutCommit` and `TimeoutPropose` parameters were moved from local config parameters into versioned parameters controlled by the state machine to ensure consistency and correctness across different protocol versions.
```

Comment:
_from [comment](https://github.com/celestiaorg/CIPs/pull/216#pullrequestreview-2373407297)_
>  I think there was some complexity involved in moving the TimeoutCommit and TimeoutPropose from local config to state-machine managed so perhaps you can elaborate on the implementation details in this CIP?

would you want to change anything here @staheri14 ?

Reply:
>I think there was some complexity involved in moving the TimeoutCommit and TimeoutPropose from local config to state-machine managed so perhaps you can elaborate on the implementation details in this CIP?

@rootulp Yes, there have been complications, we needed to change ABCI messages to include timeout info, do you mean elaborating on that aspect? or something else?

Reply:
I think he means elaborating it, but I also added line 29 after his feedback. does it look okay to you @staheri14 ?